app/models/miq_report/formatting.rb

```diff
@@ -24,57 +24,80 @@ def javascript_format(col, format_name)
-      if idx
-        col = col_order[idx]
-        format = MiqReport::Formats.details(self.col_formats[idx])
-      end
+      formatter = :_none_ if Chargeback.rate_column?(column.to_s)
     end
 
-    # Use default format for column stil nil
-    if format.nil? || format == :_default_
-      format = format_from_miq_expression(col, value)
-    else
-      format = format.deep_clone # Make sure we don't taint the original
-    end
+    formatter
+  end
 
-    options[:column] = col
+  def format_options_by(column)
+    format_options = {:column => column_to_format(column)}
+
+    formatter = formatter_by(column)
+    format_options[:format] = formatter if formatter
 
-    # Chargeback Reports: Add the selected currency in the assigned rate to options
     if Chargeback.db_is_chargeback?(db)
+      # Chargeback Reports: Add the selected currency in the assigned rate to options
       @rates_cache ||= Chargeback::RatesCache.new
-      options[:unit] = @rates_cache.currency_for_report if @rates_cache.currency_for_report
+      format_options[:unit] = @rates_cache.currency_for_report if @rates_cache.currency_for_report
     end
 
-    format.merge!(options) if format # Merge additional options that were passed in as overrides
-    value = apply_format_function(value, format) if format && !format[:function].nil?
+    format_options
+  end
+
+  def column_to_format(column)
+    if is_numeric?(column)
+      col_order[column]
+    elsif db.to_s == "VimPerformanceTrend"
+      if col == "limit_col_value"
+        db_options[:limit_col]
+      elsif col.to_s.ends_with?("_value")
+        db_options[:trend_col]
+      end
+    elsif Chargeback.db_is_chargeback?(db)
+      Chargeback.default_column_for_format(column.to_s)
+    end || column
+  end
+
+  def format_attributes_for(column_formatter, col, value)
+    format_attributes = if column_formatter == :_default_
+                           format_from_miq_expression(col, value)
+                        elsif (column_formatter.kind_of?(Symbol) || column_formatter.kind_of?(String))
+                          MiqReport::Formats.details(column_formatter)
+                        elsif column_formatter
+                          column_formatter.deep_clone # Make sure we don't taint the original
+                        elsif column_formatter.nil?
+                          # Look in this report object for column format
+                          self.col_formats ||= []
+                          idx = col_order.index(col)
+                          MiqReport::Formats.details(self.col_formats[idx])
+                        end
+
+    format_attributes || format_from_miq_expression(col, value)
+  end
+
+  def format(col, value, options = {})
+    return "" if value.nil?
+
+    options = options.merge(format_options_by(col))
+
+    column_formatter = options.delete(:format)
+    return value.to_s if column_formatter == :_none_ # Raw value was requested, do not attempt to format
```

Comment:
do we want to do the `String.new(value.to_s)` trick here too?
to be honest I don't fully understand that comment below